### PR TITLE
Feature 1

### LINKED ISSUE #1: Increment booking date automatically
always book on the date 2 days after the current day, for example: if at 0:00 on June 1st, the booking date should be June 3rd

### PR DESCRIPTION
**Pull request by BlairKH**
Fixes #1
added the function inside the handler

### CODE REVIEW

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 1]
```diff
@@ -149,4 +156,12 @@ public void book(String day, String month, String year) {
 
     }
 
+    //method increment two days
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: space after //

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 2]
```diff
@@ -149,4 +156,12 @@ public void book(String day, String month, String year) {
...
+    private LocalDateTime getDateTime() {
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: can you choose a method name that's more representative of what it does? or when it should be used

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 3]
```diff
@@ -149,4 +156,12 @@ public void book(String day, String month, String year) {
...
+        //get time zone
+        LocalDateTime today = LocalDateTime.now(ZoneId.of("America/Montreal"));
+        //return two days later
+        return today.plusDays(2);
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: are you sure that today + 2 days is the day we want to reserve? if we are on 9 june 11:59pm, will your method return june 11th or june 12th?

**Review by HankyZ — CHANGES_REQUESTED** [review]

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java (hunk 3 above)
Comment (BlairKH) [reply src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: I tested it out, it returns june 11 at 11:59 pm.

Comment (HankyZ) [reply src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: is that what we want tho?

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 4]
```diff
@@ -149,4 +158,12 @@ public void book(String day, String month, String year) {
 
     }
 
+    // method increment two days
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: for comment on a method, use /** for javadoc comment

**Review by HankyZ — CHANGES_REQUESTED** [review]

File: src/main/java/com/hankyz/ymcabooking/controller/YmcaBookingController.java [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 import com.hankyz.ymcabooking.handler.WebDriverHandler;
 
+
+
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/controller/YmcaBookingController.java]: extra line

File: src/main/java/com/hankyz/ymcabooking/controller/YmcaBookingController.java [hunk 6]
```diff
@@ -10,11 +12,10 @@ public YmcaBookingController() {
         webDriverHandler = WebDriverHandler.getInstance();
     }
 
+
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/controller/YmcaBookingController.java]: extra line

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 7]
```diff
@@ -7,6 +7,11 @@
 import org.openqa.selenium.support.ui.Select;
 import org.openqa.selenium.support.ui.WebDriverWait;
 
+import java.time.LocalDateTime;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java:14]: go to intellij->settings->editor->code style->java->imports, find "class count to use import with '*'", change the number to 30.
then run the optimize imports command

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 8]
```diff
@@ -84,8 +89,12 @@ private void signIn() {
                 .until(ExpectedConditions.presenceOfElementLocated(By.id(loggedClientFieldId)));
     }
 
-    public void book(String day, String month, String year) {
+    public void book() {
 
+        LocalDateTime bookDay = getBookingDay();
+        String day = String.valueOf(bookDay.getDayOfMonth());
+        String month = String.valueOf(bookDay.getMonthValue());
+        String year = String.valueOf(bookDay.getYear());
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: i think this can be done in the constructor

**Review by HankyZ — CHANGES_REQUESTED** [review]

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 9]
```diff
@@ -51,19 +52,40 @@
 
     private WebDriver driver;
 
+    private static LocalDateTime bookingDayTime;
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: why is this static

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 10]
```diff
@@ -14,7 +14,8 @@
 
 public class WebDriverHandler {
 
-    private final String chromeDriverPath = "drivers/chromedriver.exe";
+    private static String OS = System.getProperty("os.name").toLowerCase();
+    private final String chromeDriverPath = (OS.indexOf("mac") >= 0) ? "/usr/local/bin/chromedriver" : "drivers/chromedriver.exe";
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: do this in the constructor as well, and add mac chrome driver to project

**Review by HankyZ — CHANGES_REQUESTED** [review]

**Comment by HankyZ** [comment]
add .DS_Store to .gitignore

**Comment by HankyZ** [comment]
set the driver paths to drivers/macos/chromedriver and win too

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 11]
```diff
@@ -7,9 +7,15 @@
...
 
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+
 public class WebDriverHandler {
 
-    private final String chromeDriverPath = "drivers/chromedriver.exe";
+    private final String OS = System.getProperty("os.name").toLowerCase();
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java]: this is not needed as a global var, a local var suffice

File: src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java [hunk 12]
```diff
@@ -46,19 +52,42 @@
...
 
+    private final LocalDateTime bookingDayTime;
+
     public static WebDriverHandler getInstance() {
         if (instance == null)
             instance = new WebDriverHandler();
         return instance;
     }
 
+    /**
```
Comment (HankyZ) [inline src/main/java/com/hankyz/ymcabooking/handler/WebDriverHandler.java:74]: let's put this after the constructor

**Review by HankyZ — CHANGES_REQUESTED** [review]

**Review by HankyZ — APPROVED** [review]

**Pull request merged**